src/lib/process/process.c

```diff
@@ -0,0 +1,521 @@
+/* Copyright (c) 2003, Roger Dingledine
+ * Copyright (c) 2004-2006, Roger Dingledine, Nick Mathewson.
+ * Copyright (c) 2007-2018, The Tor Project, Inc. */
+/* See LICENSE for licensing information */
+
+/**
+ * \file process.c
+ * \brief Module for working with other processes.
+ **/
+
+#define PROCESS_PRIVATE
+#include "lib/container/buffers.h"
+#include "lib/net/buffers_net.h"
+#include "lib/container/smartlist.h"
+#include "lib/log/log.h"
+#include "lib/log/util_bug.h"
+#include "lib/process/process.h"
+#include "lib/process/process_unix.h"
+#include "lib/process/process_win32.h"
+
+#include <stddef.h>
+
+static smartlist_t *processes;
+
+struct process_t {
```

Comment:
The `typedef` can be found in `process.h`. All the code here tries to hide the members from the struct for other modules, but because the `typedef` is still in place we can continue to use `process_t` instead of `struct process_t`.